api/base/v1alpha1/vectorstore_types.go

```diff
@@ -29,13 +29,28 @@ type VectorStoreSpec struct {
 	Endpoint *Endpoint `json:"endpoint,omitempty"`
 
 	Chroma *Chroma `json:"chroma,omitempty"`
+
+	PGVector *PGVector `json:"pgvector,omitempty"`
 }
 
 // Chroma defines the configuration of Chroma
 type Chroma struct {
 	DistanceFunction chromago.DistanceFunction `json:"distanceFunction,omitempty"`
 }
 
+type PGVector struct {
+	// PreDeleteCollection defines if the collection should be deleted before creating.
+	PreDeleteCollection bool `json:"preDeleteCollection,omitempty"`
+	// CollectionName defines the name of the collection
+	CollectionName string `json:"collectionName,omitempty"`
```

Comment:
What are the differences between  `CollectionName  `  and  `CollectionTableName `  ?

Reply:
https://github.com/langchain-ai/langchain/blob/85a4594ed7dbde9d257e230d5d789b68c42421cf/libs/community/langchain_community/vectorstores/pgvector.py#L154C27-L154C27

These are configurables introduced by langchain python, which langchain-go replicates, and is just exposing them for advanced users to use here.
Normal users don't need to configure it, just use the default value is enough.